gobblin-core/src/main/java/gobblin/converter/jdbc/AvroToJdbcEntryConverter.java

```diff
@@ -0,0 +1,354 @@
+    init(workUnit);
+  }
+
+  /**
+   * Fetches JdbcWriterCommands.
+   * Builds field name mapping between Avro and JDBC.
+   * {@inheritDoc}
+   * @see gobblin.converter.Converter#init(gobblin.configuration.WorkUnitState)
+   */
+  @Override
+  public Converter<Schema, JdbcEntrySchema, GenericRecord, JdbcEntryData> init(WorkUnitState workUnit) {
+    this.commands = jdbcWriterCommandsFactory.newInstance(workUnit);
+
+
+    String avroToJdbcFieldsPairJsonStr = workUnit.getProp(ConfigurationKeys.CONVERTER_AVRO_JDBC_ENTRY_FIELDS_PAIRS);
+    if(!StringUtils.isEmpty(avroToJdbcFieldsPairJsonStr)) {
+      if (!avroToJdbcColPairs.isPresent()) {
+        ImmutableMap.Builder<String, String> avroToJdbcBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<String, String> jdbcToAvroBuilder = ImmutableMap.builder();
+
+        JsonObject json = new JsonParser().parse(avroToJdbcFieldsPairJsonStr).getAsJsonObject();
+        for (Map.Entry<String, JsonElement> entry : json.entrySet()) {
+          if (!entry.getValue().isJsonPrimitive()) {
+            throw new IllegalArgumentException("Json value should be a primitive String. "
+                + ConfigurationKeys.CONVERTER_AVRO_JDBC_ENTRY_FIELDS_PAIRS + " : " + avroToJdbcFieldsPairJsonStr);
+          }
+          avroToJdbcBuilder.put(entry.getKey(), entry.getValue().getAsString());
+          jdbcToAvroBuilder.put(entry.getValue().getAsString(), entry.getKey());
+        }
+        avroToJdbcColPairs = Optional.of((Map<String, String>) avroToJdbcBuilder.build());
+        jdbcToAvroColPairs = Optional.of((Map<String, String>) jdbcToAvroBuilder.build());
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Converts Avro schema to JdbcEntrySchema.
+   *
+   * Few precondition to the Avro schema
+   * 1. Avro schema should have one entry type record at first depth.
+   * 2. Avro schema can recurse by having record inside record. As RDBMS structure is not recursive, this is not allowed.
+   * 3. Supported Avro primitive types and conversion
+   *  boolean --> java.lang.Boolean
+   *  int --> java.lang.Integer
+   *  long --> java.lang.Long or java.sql.Date , java.sql.Time , java.sql.Timestamp
+   *  float --> java.lang.Float
+   *  double --> java.lang.Double
+   *  bytes --> byte[]
+   *  string --> java.lang.String
+   *  null: only allowed if it's within union (see complex types for more details)
+   * 4. Supported Avro complex types
+   *  Records: Only first level depth can have Records type. Basically converter will peel out Records type and start with 2nd level.
+   *  Enum --> java.lang.String
+   *  Unions --> Only allowed if it have one primitive type in it or null type with one primitive type where null will be ignored.
+   *  Once Union is narrowed down to one primitive type, it will follow conversion of primitive type above.
+   * {@inheritDoc}
+   *
+   * 5. In order to make conversion from Avro long type to java.sql.Date or java.sql.Time or java.sql.Timestamp,
```

Comment:
I assume this metadata doesn't change between WorkUnits, so can it just be determined once in the driver and stored somewhere (e.g. the WorkUnit). This avoids having some to make an extra JDBC connection in the `Converter`.